lib/mongoid/core_ext/relations/options.rb

```diff
@@ -1,7 +1,21 @@
 module Mongoid
   module Relations
     module Options
-      COMMON << :versioned
+      VERSIONED_OPTIONS  = [:versioned].freeze
+
+      def validate!(options)
+        valid_options = options[:relation]::VALID_OPTIONS + COMMON + VERSIONED_OPTIONS
```

Comment:
@ebeigarts yes, looks definetly cleaner, only in case of invalid option exception message contains all valid options, but in this case versioned wont be included in exception message.